main.js

```diff
@@ -1 +1,96 @@
-var x;
+var Product = function () {
+    this.w = 0;
+    this.h = 0;
+};
+Product.prototype.render = function () {
+    var node = document.createElement('div');
+    node.setAttribute('style', 'border: 1px solid #000000; padding: 2px; margin: 2px; width:'
+        + this.w + 'px; height: ' + this.h + 'px; background-color: ' + this.bg_colour);
+    node.innerText = this.name;
+    return node;
+}
+
+var Treatment = function () {
+    var that = new Product();
+    that.w = 120;
+    that.h = 28;
+    return that;
+}
+Treatment.treatmentMax = 100;
+
+
+/**
+ * Sort the contents of a menu in either accending or decending (depending on
+ * flag set) greyscale colour order. NOTE, this will not work with non-greyscale
+ * colurs so don't use it for that!
+ *
+ * @param bool asc true for assending, false for not
+ */
+function bgSort(input, asc) {
+    // Rearange the contents of the menu based on the bg colour
+    var tmp = [];
+    var asc = asc || false;
+
+    for (i = 0; i < input.length; i++) {
+        for (j = 0; j < input.length; j++) {
+            var thisColour = parseInt(input[i].bg_colour.replace("#", ''), 16),
+                nextColour = parseInt(input[j].bg_colour.replace("#", ''), 16);
+
+            if (asc) {
+                if (thisColour < nextColour) {
+                    // Switch them around
+
+                    tmp = input[i];
+                    input[i] = input[j];
+                    input[j] = tmp;
+                }
+            }
+            else {
+                if (thisColour > nextColour) {
+                    // Switch them around
+
+                    tmp = input[i];
+                    input[i] = input[j];
```

Comment:
this is doing exactly the same as the above. No need for an if/else block in that case.